### PR TITLE
#736: Button colors

### PR DESCRIPTION
**Pull request by hani-ibrahim**
fixing #736 

finally finished this Ticket 😤

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasSDK/Commons/Array.swift [hunk 1]
```diff
@@ -14,3 +14,12 @@ extension Array {
     }
 
 }
+
+extension Array where Element: Equatable {
+
+    mutating func remove(item: Element) {
+        self = self.filter { $0 != item }
+    }
```
Comment (bonkey) [inline AtlasSDK/Commons/Array.swift]: 👍

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift [hunk 2]
```diff
@@ -38,31 +39,41 @@ class CheckoutSummaryEditProductStackView: UIStackView {
...
     dynamic fileprivate func sizeButtonTapped() {
-        guard let selectedArticle = dataSource?.selectedArticle else { return }
-        dataSource?.checkoutContainer.displaySizes(selectedArticle: selectedArticle, animated: true) { [weak self] idx in
+        guard let selectedArticle = dataSource?.selectedArticle, let checkoutContainer = dataSource?.checkoutContainer else { return }
+        let type = checkoutContainer.displaySizes(selectedArticle: selectedArticle, animated: true, hiddenHandler: { [weak self] in
+            self?.updateArrow(for: nil, animated: true)
+        }, completion: { [weak self] idx in
             self?.sizeSelected(at: idx, for: selectedArticle)
-        }
+        })
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: kf291wlcf7y5a

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift [hunk 3]
```diff
@@ -38,31 +39,41 @@ class CheckoutSummaryEditProductStackView: UIStackView {
...
     dynamic fileprivate func quantityButtonTapped() {
-        guard let selectedArticle = dataSource?.selectedArticle else { return }
-        dataSource?.checkoutContainer.displayQuantites(selectedArticle: selectedArticle, animated: true) { [weak self] idx in
+        guard let selectedArticle = dataSource?.selectedArticle, let checkoutContainer = dataSource?.checkoutContainer else { return }
+        let type = checkoutContainer.displayQuantites(selectedArticle: selectedArticle, animated: true, hiddenHandler: { [weak self] in
+            self?.updateArrow(for: nil, animated: true)
+        }, completion: { [weak self] idx in
             self?.quantitySelected(at: idx, for: selectedArticle)
-        }
+        })
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: Like _good_ **old** objective-c

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift [hunk 4]
```diff
@@ -83,6 +94,36 @@ class CheckoutSummaryEditProductStackView: UIStackView {
...
+            imageViewsToClose.remove(item: imageView)
+        }
+
+        imageViewsToClose.forEach { updateArrow(arrowImageView: $0, opened: false, animated: animated) }
+    }
+
+    private func updateArrow(arrowImageView: UIImageView, opened: Bool, animated: Bool) {
+        let transform = opened ? CGAffineTransform(rotationAngle: CGFloat.pi) : CGAffineTransform.identity
+        if animated {
+            UIView.animate(duration: .fast) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: Wouldn't `duration: animated ? 0 : .fast` work as good as this nasty if?

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift (hunk 3 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: There is somehow a rule that I read before (which I think is fine): If a function has 2 blocks then name the 2 of them for readability.

For example on UIView animation, Where this one:

```swift
UIView.animate(withDuration: 1, animations: {
   // Animation
}, completion: {  _ in
   // Completion
})
```

Is much more readable than this one:

```swift
UIView.animate(withDuration: 1, animations: {
   // Animation
}) {  _ in
   // Completion
}
```

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift (hunk 4 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: actually I thought about that but didn't want to add overhead to animate in zero time, But after [some search](http://stackoverflow.com/a/15492533/1722480), seams to be okay

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift (hunk 3 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: I'm not about naming, I'm about congestion.

More readable would be:

```swift
let animations = {
// Animation
}
let completion = {
// Completion
}
UIView.animate(withDuration: 1, animations: animations, competion: completion)
```

or 

```swift
// ...
UIView.animate(withDuration: 1, animations: animations) {
}
```

And you could pass functions to it as well:

```swift
func formatString(str: String) -> String {
    return "formatString: \(str)"
}

func formatInt(int: Int) -> String {
    return "formatInt: \(int)"
}

func printIntString(stringf: (_ string: String) -> String, intf: (_ int: Int) -> String) {
    let s = "STRING"
    let i = 42
    print("string: (\(stringf(s))), int (\(intf(i)))")
}

printIntString(stringf: { str in
    return "inline: \(str)"
}, intf: { int in
    return "inline: \(int)"
})

printIntString(stringf: formatString, intf: formatInt)

// return:
// string: (inline: STRING), int (inline: 42)
// string: (formatString: STRING), int (formatInt: 42)


```

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/Utils/UIView.swift [hunk 5]
```diff
@@ -151,10 +151,19 @@ extension UIView {
...
+            return
+        }
         UIView.animate(withDuration: duration.rawValue, animations: animations)
     }
 
     static func animate(duration: AnimationDuration = .default, animations: @escaping () -> Void, completion: @escaping (Bool) -> Void) {
+        guard duration != .noAnimation else {
+            animations()
+            completion(true)
+            return
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/UIView.swift]: @bonkey I'm not sure about this

I tried and it works, but still don't like to set a time with zero 👹 
```
       UIView.animate(withDuration: 0, animations: {
            print("Animation")
        }) { animated in
            print("Completion")
        }
```

**Comment by bonkey** [comment]
👍

**Pull request merged**